internal/tools/pipeline/providers/cron/daemon/provider_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/coreos/etcd/clientv3"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle"
+	db2 "github.com/erda-project/erda/internal/tools/pipeline/providers/cron/db"
+	"github.com/erda-project/erda/internal/tools/pipeline/providers/edgepipeline_register"
+)
+
+type EdgePipelineRegister struct {
```

Comment:
use internal/tools/pipeline/providers/edgepipeline_register/mock.go:25

Reply:
The return value of the mock is different and requires a separate mock

Reply:
use embed and overwrite some needed methods.

Reply:
done